### ISSUE TITLE
Auto Download No Longer Working

### ISSUE THREAD

**Issue by timwithtots** [SchizoDuckie/DuckieTV#1387]
**What build of DuckieTV are you using (Standlone / Chrome Extension (New Tab / Browser Action))**
Standalone

**What version of DuckieTV are you using (Stable 1.1.x / Nightly yyyymmddHHMM)**
 nightly v202303121925

**What is your Operating System (Windows, Mac, Linux, Android)**
Windows

**Describe the problem you are having and steps to reproduce if available**
Over the last month Auto-download seems to work sporadically. Generally speaking auto-download will work maybe 2 of 20 times I load Duckie. Under statistics it states [Auto Download Service (enabled but not active)

**Attach any DuckieTV statistics or Developer Console logs if available**

**Comment by garfield69** [comment]
On your Dtv page press **F12** to open the _development tools_ page, and click on the **Console** _tab_ to view the log and look for error messages related to your torrent client.
Also use the **ADSM** to get a view of the Auto-download activity to see what is happening when ADS tries to schedule a download for an episode.
Here are a few links to wiki pages that may give you better insight to ADS, ADSM and configuring and connecting Dtv to your torrent client.
https://github.com/SchizoDuckie/DuckieTV/wiki/Auto-Download-FAQs
https://github.com/SchizoDuckie/DuckieTV/wiki/Auto-Download-FAQs#auto-download-service-monitor
https://github.com/SchizoDuckie/DuckieTV/wiki/Setting-up-Torrent-Clients-for-DuckieTV

if you need further help diagnosing the issue, then start by providing details on which torrent client you are using, its version, so I can cross test to verify that the torrent client you are using has not had any updates that require Dtv support changes.
Let me know how you get on.

**Comment by timwithtots** [comment]
Appreciate the prompt reply! I tried a full re-install (deleted the Duckie folder prior) and now Duckie doesn't want to load properly. The program launches however the calendar is completely blank if I load it AFTER I launch/start my VPN service (Nord). If I launch it prior to starting Nord the calendar is working just fine however all other data fails to load. Totally separate issue but I wanted to call this out. This didn't start until after the re-install.

With Nord disabled/not running I'm still having issues with auto-download and loading Duckie in general. The audo-download icon on the left of Duckie displays a yellow dot and when I click anything on the left menu bar everything just shows "loading".

For my torrent client I'm using qBittorrent v4.5.2 (64-bit)

I'm seeing the following errors:

Failed to load resource: net::ERR_CONNECTION_REFUSED
Failed to load resource: net::ERR_CERT_AUTHORITY_INVALID
Uncaught ReferenceError: Loggr is not defined


I've been using Duckie for about 3-4 years now I believe and this is really the first time I've run into an issue. Thank you for an amazon product and this great support. Any help is much appreciated!

**Comment by garfield69** [comment]
Urg. OK, lots of factors at play here.
I don't have nordVPN, but I presume this is a system level VPN, which is to say, that once started all traffic generated on that windows PC is routed in and out of the VPN.
So for a test on my win10 sandbox, 
- I started ProtonVPN, established a connection to a server in the USA, 
- and once connected, started qBt 4.5.2
- and then started Dtv standalone nightly 202303121925

Dtv takes some time to connect to qBt 4.5.2, as the first attempt it tries always fails, but the second attempt within the next 30 seconds usually connects.
Then ADS launches, uses my default Search Engine (in this case its TGx) to find an episode and begins to download. I use the ADSM to check that the episode was found and sent to qBt and I check the client to see that a torrent is active.

So I know in principle, that it is possible to run Dtv, qBt and a system VPN, so any problems that arise are all down to your system setup and VPN, qBt or Dtv settings.

Lets try to first establish that Dtv is working.
- shutdown your VPN, and qBt
- start Dtv and access **Dtv>Settings>autoDownload Torrents** and click disable, we don't want ADS active right now.
- check that the calendar loads, and wait for the trakt.tv episodes updates to complete.
- if you can navigate to view series and episodes and the About page without issues, then we should be happy that Dtv is ok.
- now shutdown Dtv 
- start qBt. We are going to check that Dtv can connect to qBt.
- start Dtv and wait for it to connect to qBt, which should take no more that 30 seconds. If connected you should see the Dtv client icon go from a red dot, to orange to green.
- If that all works then the next step is to introduce the VPN
- shutdown Dtv, and qBt
- start up nordVPN and establish a server connection
- start up qBt
- startup Dtv
- after 30 seconds check that Dtv is connected to qBt
- next, test your default Dtv search engine by doing a manual search for an episode and getting search results on your torrent search dialogue.
- if searching for torrents is working then we can go back to settings and enable ADS.
- ADS should start looking for episodes and sending torrent to qBt to manage downloads.

Its important to remember that the sequence in which you start up the VPN, qBt and Dtv is important, and if VPN is shutdown or is disconnected, then you have to shutdown qBt and Dtv too, before restarting the VPN sessions again.
Also, Dtv should be started last. If you shutdown qBt then you need to shutdown Dtv too.

Let me know how you get on.

**Comment by timwithtots** [comment]
Confirmed VPN/QBT were closed and nothing was running via Task Manager.  Once confirmed I started going down the list of steps you provided and I got stuck pretty early on.

- check that the calendar loads, and wait for the trakt.tv episodes updates to complete.

I loaded DTV > disabled ADS and trakt.tv started it's scan and got stuck on the final scan. In my case this was ~136/136. It was stuck for ~5 minutes and any time I clicked anywhere else in DTV it would just show "loading". I ended up force closing to restart DTV and now when launching DTV the calendar shows as being accurate however any time I click on anything I just see "loading" on the righthand side. 

Please let me know if you want the remaining steps checked and I'll go down the list.

**Comment by garfield69** [comment]
Interesting. 
- So with just Dtv running as before, press **F12** and click on the **console** tab.
- on the line that has the _Elements, Console, Sources_ tabs etc, click on the **settings** icon
- scroll down to the _console_ section and tick the **show timestamps** checkbox
- close the settings page to get back to the console log
- on the line that has the **filter** box, to its right is a pull down for the **levels**, which is probably set to _Default Levels_.
- tick all the levels (Verbose, Info, Warnings, Errors)
- scroll to the top of the log and on a _blank space_ close to the quack! **right click** to get the **options** and select the **save as** option
- click the save button and note where that log has been saved, (probably in your downloads folder)
- **drag/drop** that file here so I can have a look and see if I can spot anything

**Comment by timwithtots** [comment]
Hoping it's just something simple....

[bagblfnohpiddokbpaacecffdboomafe-1682107900604.log]()

**Comment by garfield69** [comment]
It looks like you have the **Opt-In Error tracking** _enabled_ (its a button at the bottom of the Dtv About page), but since you cannot access that service (the certificate at api.loggr.net/ appears to be expired so that service is not available ATM), so you are getting net::ERR_CERT_AUTHORITY_INVALID, which is crashing the Dtv JavaScript.
I just tested this on my sandbox and I get the same issue now.

A little tricky to fix, but I managed it, so you can try to have a go too. The key is to get in quick as soon as you start Dtv, and switch off that option.
So 
- start Dtv,
- click on the (i) About icon to open the About Dtv side panel,
- scroll to the bottom,
- and click the disable Opt-In Error Tracking option.

if you get in early enough it should allow you to navigate the rest of the Dtv pages.
If you cannot get it turned off then I will need to look at how we can turn it off some other way, probably with a command from the console log, but I hope we don't have to go there.

**Comment by timwithtots** [comment]
Okay...I was able to get in quick enough and disable. I remember I enabled that because of the ADS issue.

DTV is loading just fine both with/without VPN running prior. I enabled ADS and yesterdays shows kicked off but anything prior didn't. Just a quick call out...I had it set to 4 days then moved it to 10 recently trying to fix ADS not working so its currently set to 10. To try and push the ADS for anything prior to yesterday I changed the Default Search Provider to TGx and the pending torrents prior to yesterday kicked off.

If it's cool with you I'd like to keep this open to track this over the next few days to see how ADS works going forward.

To summarize my current settings FWIW:

Default Search Provider: TGx
Auto DL Period: 10 days
Auto DL Delay: 15 minutes (I believe I changed this awhile ago when shows started becoming widely available on streaming services earlier than the original air time.

I should have a better idea how ADS is working by Mon/Tues

Thanks again for the awesome support!

**Comment by garfield69** [comment]
That is good news.
Yes by all means monitor and report back. I'll close the ticket in a week if I haven't heard anything before then.
I've done some digging into the log_GR service that we've been using since 2017, it seems that they are no longer providing that log tracking service, so I shall be releasing an update with that option removed.

**Comment by ggervais5** [comment]
I too, am using the same version of Dtv on Windows Pro (64 bit).  Since I installed this version, auto download doesn't find anything, yet when I manually search, I do find the same shows.  My default provider is 1337x, which I've used for eons with great success, until now.   I've tried the workaround posted on Reddit, but it doesn't work (the service runs, but it doesn't find torrents) with this version.

**Comment by timwithtots** [comment]
FWIW I also use 1337 for certain shows and can confirm I’ve experienced this issue with that site as well. I’m also using Pro 64 bit.

**Comment by garfield69** [comment]
FYI
Yes, the web site 1337x.to is not finding any of the shows on my current weeks schedule, so this is not a problem with Dtv but with the 1337x web site and there is nothing I can do about it.
If you have turned on the Dtv multi-search-engine manual search, you can find which search engine returns the most results and make that your default search engine.
I use TGx currently.
The ADS only uses the default search engine, or optionally an alternate search engine for a specific series if you set that override in that series settings.

**Comment by garfield69** [comment]
re: 1337x.to you may be interested in https://torrentfreak.com/1337xs-search-is-broken-cant-find-recent-torrents-230425/

**Comment by ggervais5** [comment]
Interesting.  I switched the primary torrent engine yo TgX and it appears
no better than 1337x for auto downloading.  I'm testing Jackett with
DuckieTV to see if I can get this working again.

**Comment by garfield69** [comment]
> I'm testing Jackett with DuckieTV

good idea.
Most sites seems to be activating cloudflare challenges these days to protect against DDoS attacks and whatnot, so its possible that the Dtv native search engines are not working because you are getting  cloudflare challenges that Dtv cannot resolve. 
 Jackett in conjunction with the flaresolverr proxy can handle most of the simple cloudflare challenges, so using Dtv with the Jackett indexers could be a way in.

**Comment by ggervais5** [comment]
I got the idea from Sonarr.  Are you still looking for developers to help
out with Dtv?  I know one with Node.js experience.

I am a c# developer, which doesn't help.

Greg

**Comment by garfield69** [comment]
There is no node.js code in Dtv, it is all AngularJS 1.7.3
BTW it appears that 1337x have fixed their issue now, the spotlight from torrentfreak must have jolted someone into action ;-D

**Comment by timwithtots** [comment]
Just wanted to provide an update - It seems ADS is working just fine now. I'm not sure if it was the re-install, enabling multi-torrent search, etc....but either way I'm good to go on my end. Thanks again for the support and getting this all squared away.

**Comment by ggervais5** [comment]
Oddly enough, ADS started working for me too, without reinstalling, or any of the things timewithtots did.  I suspect the timing was coincidental - the issue is likely to be at the Torrent site end, not within DuckieTV.  For me, ADS stopped work, and started working on its own.
